algorithm.go

```diff
@@ -161,10 +169,13 @@ func (a Algorithm) computeHash(data []byte) ([]byte, error) {
 // set to 0, no hash is used for this algorithm.
 // The parameter `hashFunc` is preferred in the case that the hash algorithm is not
 // supported by the golang built-in crypto hashes.
+// It is safe for concurrent use by multiple goroutines.
```

Comment:
Changes looks good! Just a side question. 

Should we not allow user to De-Register an External Algorithm and add a new one. I do not see any API like that?